arch/lkl/Makefile.postlink

```diff
@@ -0,0 +1,53 @@
+# The post-link vmlinux pass is only relevant when LKL is being built with
+# libFuzzer instrumentation (-fsanitize=fuzzer). SanitizerCoverage
+# emits GRP_COMDAT sections in the generated object files to group sections
+# with instrumented code, coverage counters and relocations respectively.
+# The linker semantics for incremental linking with section groups is unclear
+# and results in the link-time errors. This extra post-link pass implements a
+# workaround for this problem:
+#
+# 1) Vmlinux is incrementally linked with an empty linker script. Thus, the input
+#    sections aren't merged in the generated vmlinux image (ld and lld do still
+#    merge some common sections such as .text. .rela.text, .data and etc but it
+#    doesn't casue a problem).
+#
+# 2) Remove GRP_COMDAT group sections with name .group from the original vmlinux.
+#
+# 3) Relink the stripped vmlinux using the original LKL linker script.
+#    As there is no group sections in the input object, merging sections doesn't
+#    cause any issues.
+#
+# TODO: The above steps aren't solution to the problem but rather a short-term
+# workaround. At the moment removing .group sections from the incrementally
+# linked vmlinux doesn't seem to cause any issues as the only GRP_COMDAT sections
+# present in the object files are the ones emited by SanitizerCoverage
+# instrumentation. However, here is a couple of other things to take into
+# consideration:
+#
+#  * --gc-sections linker option could remove __sancov_pcs sections, thus, this
+#    option shouldn't be used when building fuzzers.
+#
+#  * If the linker removes a function there might be 'dangling' __sancov_xxx
+#    sections. However, this isn't an issue for fuzzing.
+#
+#  * __sancov_pcs and __sancov_cntrs sections need to be merged in the corresponding
+#    order (so that the value of the counters match to values of PC of their
+#    corresponding basic blocks). With this workaround this seem to be the case
+#    (while the linker might not guarantee this). However, even if these sections
+#    aren't merged in the order, technically, this won't be a problem for fuzzing.
+ifneq ($(LKL_FUZZING),)
```

Comment:
I wonder if should should be a more specific config option like CONFIG_LLVM_FUZZER that we can enable when LLVM fuzzing is enabled. Although right now LLVM fuzzing is the only option that might change in the future.

Definitely not a big deal, feel free to disregard this, we can easily change it if/when we support gcc based fuzzing.

Reply:
Yes, I believe this is specific to LLVM libFuzzer implementation (more specifically to SanitizerCoverage instrumentation in LLVM toolchain) and other fuzzing engines might not be affected to this issue. If you are OK with it, I would leave it like this and change when needed as you suggested above..

Reply:
Thanks for confirming Eugene. Sounds good, will refactor when/if needed.